diff --git a/ofstd/include/dcmtk/ofstd/oftypes.h b/ofstd/include/dcmtk/ofstd/oftypes.h
index 217ec33..1697c4e 100644
--- a/ofstd/include/dcmtk/ofstd/oftypes.h
+++ b/ofstd/include/dcmtk/ofstd/oftypes.h
@@ -106,7 +106,7 @@ typedef double          Float64;    /* 64 Bit Floating Point Double */
 #define OFulonglong ulonglong
 #endif
 
-#ifdef HAVE_INT64_T
+#if defined(HAVE_INT64_T) && !defined(_WIN32)
 /* many platforms define int64_t in <cstdint> */
 typedef int64_t       Sint64;
 #elif SIZEOF_LONG == 8
@@ -120,7 +120,7 @@ typedef OFlonglong    Sint64;
 #define OF_NO_SINT64 1
 #endif
 
-#ifdef HAVE_INT64_T
+#if defined(HAVE_INT64_T) && !defined(_WIN32)
 /* many platforms define uint64_t in <cstdint> */
 typedef uint64_t      Uint64;
 #elif SIZEOF_LONG == 8
